include/PatternStore.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * PatternStore.h - model-component of Pattern Editor
+ *
+ * Copyright (c) 2004-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#ifndef PATTERN_STORE_H
+#define PATTERN_STORE_H
+
+#include "TrackContainer.h"
+#include "ComboBoxModel.h"
+
+
+/*
+ * PatternStore is the backend of Pattern Editor:
+ *
+ * +----------------------------------+
+ * | PatternStore (TrackContainer)    |
+ * |                                  |
+ * | +------------------------------+ |
+ * | | Track 1   [Clip A]  [Clip B] | |
+ * | +------------------------------+ |
+ * |                                  |
+ * | +------------------------------+ |
+ * | | Track 2   [Clip C]  [Clip D] | |
+ * | +------------------------------+ |
+ * +---------------------------------+
+ *
+ * Pattern Editor displays one pattern at a time, but they all belong to the same PatternStore.
+ * A "pattern" is not an object, it's just a word we use to describe a column of clips.
+ * In the illustration above, Clip A and Clip C would be "Pattern 1".
+ *
+ * Do not confuse the Tracks in the PatternStore with PatternTracks.
+ * PatternTracks are used in the Song Editor to reference a pattern.
```

Comment:
I still miss `PatternClip` in this diagram. Possible change:

```suggestion
 * PatternTracks are used in the Song Editor and contain PatternClips that each reference a pattern.
```